src/hotspot/share/opto/node.cpp

```diff
@@ -903,10 +911,17 @@ void Node::disconnect_inputs(Compile* C) {
 
   // Remove precedence edges if any exist
   // Note: Safepoints may have precedence edges, even during parsing
-  for (uint i = req(); i < len(); ++i) {
-    set_prec(i, nullptr);
+  for (uint i = len() - 1; i < len() && i >= req(); --i) {
+    rm_prec(i);                // no-op if _in[i] is nullptr
   }
 
+#ifndef PRODUCT
```

Comment:
This should be #ifdef ASSERT

Reply:
got it.  I will update it in next revision.